spec/system/provider_interface/provider_changes_an_offer_spec.rb

```diff
@@ -41,13 +44,30 @@ def and_i_am_permitted_to_see_applications_for_two_providers
   end
 
   def and_an_offered_application_choice_exists_for_one_of_my_providers
-    @course_option_one = course_option_for_provider_code(provider_code: @provider.code)
+    @course_option_one = course_option_for_provider(provider: @provider, study_mode: 'full_time')
     @application_offered = create(:application_choice, :with_offer, course_option: @course_option_one)
   end
 
-  def and_another_two_course_options_exist_for_this_provider
-    @course_option_two = course_option_for_provider_code(provider_code: @provider.code)
-    @course_option_three = create(:course_option, course: @course_option_two.course, site: create(:site, provider: @provider))
+  def and_another_relevant_course_exists_for_this_provider
+    @other_course = create(:course, :open_on_apply, :with_both_study_modes, provider: @provider)
+    @course_option_two = create(:course_option, :full_time, course: @other_course)
+    @course_option_three = create(:course_option, :part_time, course: @other_course)
+  end
+
+  def and_a_part_time_only_course_exists_for_this_provider
+    @part_time_course = create(:course, :open_on_apply, :part_time, provider: @provider)
+  end
+
+  def and_an_older_course_exists_for_this_provider
```

Comment:
Nice to see this here! Hoping we can have a proper recruitment cycle kickoff very soon